rdt/transformers/numerical.py

```diff
@@ -184,17 +184,18 @@ def _reverse_transform(self, data):
         data = self.null_transformer.reverse_transform(data)
         if self.enforce_min_max_values:
             data = data.clip(self._min_value, self._max_value)
-        elif self.computer_representation != 'Float':
+        elif not self.computer_representation.startswith('Float'):
             min_bound, max_bound = INTEGER_BOUNDS[self.computer_representation]
             data = data.clip(min_bound, max_bound)
 
-        is_integer = np.dtype(self._dtype).kind == 'i'
+        is_integer = pd.api.types.is_integer_dtype(self._dtype)
+        is_pandas_instance = isinstance(data, (pd.Series, pd.DataFrame))
         if self.learn_rounding_scheme and self._rounding_digits is not None:
             data = data.round(self._rounding_digits)
         elif is_integer:
             data = data.round(0)
 
-        if pd.isna(data).any() and is_integer:
+        if pd.isna(data).any() and is_integer and not is_pandas_instance:
```

Comment:
How can it be an integer type, have nans and not be pandas? Numpy doesn't support nans so if it isn't a pandas object doesn't that mean it has to either be an int type or have no nans?

Reply:
Yes, you're correct. I clarified the statement in [02f0c1a](https://github.com/sdv-dev/RDT/pull/857/commits/02f0c1aa001b046c347323cffedb07cd526deab4).

There is a case where the input that was fitted was a `numpy ndarray` of `dtype int` (therefore without `NaNs`).
However, the input data that we want to `reverse_transform` from is a `np.ndarray of float with NaNs`. In this case we still have to return a `numpy ndarray of float` in order to have `NaNs`, this is what the following test expects (the test is failing with if `integer dtype` is returned):
https://github.com/sdv-dev/RDT/blob/dd654c4911e13f95a6afe70d4d34f4013bd4f237/tests/unit/transformers/test_numerical.py#L461


Nevertheless, I don't know if this case makes sense from the SDV side because it could happen that SDV generated NaNs for an integer column that didn't have any during the fitting, but because it's a  `pandas` instance and not `numpy array`, everything should be fine.

Reply:
is_integer is calculated on the data though, so if the data is an np.ndarray of floats, is_integer would be False no?

Reply:
`is_integer` is calculated on the dtype of the data that was fitted by the transformer:
https://github.com/sdv-dev/RDT/blob/02f0c1aa001b046c347323cffedb07cd526deab4/rdt/transformers/numerical.py#L191

Reply:
ah I see. Ok